tests/contrib/test_viz.py

```diff
@@ -1,12 +1,78 @@
-import pytest
 import json
+
+import pytest
+from matplotlib.figure import Figure
+
 import pyhf.contrib.viz.brazil as brazil
-import matplotlib.pyplot as plt
+
+# Tests with the @pytest.mark.mpl_image_compare decorator compare against
+# reference images generated via:
+# pytest --mpl-generate-path=tests/contrib/baseline tests/contrib/test_viz.py
 
 
 @pytest.mark.mpl_image_compare
-def test_brazil():
-    data = json.load(open('tests/contrib/hypotestresults.json'))
-    fig, ax = plt.subplots(1, 1)
+def test_plot_results():
+    data = json.load(open("tests/contrib/hypotest_results.json"))
+
+    fig = Figure()
+    ax = fig.subplots()
     brazil.plot_results(ax, data['testmus'], data['results'], test_size=0.05)
     return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_plot_cls_components():
+    data = json.load(open("tests/contrib/tail_probs_hypotest_results.json"))
+
+    fig = Figure()
+    ax = fig.subplots()
+    brazil.plot_cls_components(ax, data["testmus"], data["results"], test_size=0.05)
+    return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_plot_cls_components_no_clb():
+    data = json.load(open("tests/contrib/tail_probs_hypotest_results.json"))
```

Comment:
you should use the `testdata` fixture in this case. You'd just need a `tests/contrib/test_viz` folder with the data, and it will point at the right path for you and load it up correctly.

Reply:
Good point. Fixed it.